docs/style.md

```diff
@@ -0,0 +1,188 @@
+
+Reason: Ruby code is executed on the system that runs InSpec. This allows
+InSpec to work without Ruby and rubygems being required on remote
+targets (servers or containers). System calls are often used to interact with
+the local OS or remote endpoints from a local installation.
+InSpec tests, however, are designed to be universally executable on all
+types of runtimes, including local and remote execution. We want to give
+users the ability to take an OS profile and execute it remotely or locally.
+
+**Avoid shelling out**
+
+Avoid: `` `ls``\`
+Avoid: `system("ls")`
+Avoid: `IO.popen("ls")`
+Use: `command("ls")` or `powershell("..")`
+
+Ruby's command executors will only run localy. Imagine a test like this:
+
+```ruby
+describe `whoami` do
+  it { should eq "bob\n" }
+end
+```
+
+If you run this test on your local system and happen to be using Bob's account
+it will succeed. But if you were to run it against `--target alice@remote-host.com`
+it will still report that the user is bob instead of alice.
+
+Instead, do this:
+
+```ruby
+describe command('whoami') do
+  its(:stdout) { should eq "bob\n" }
+end
+```
+
+If the profile is pointed to a remote endpoint using the `command` resource
+will run it on the remote OS.
+
+**Avoid Ruby IO on files**
+
+Avoid: `File.new("filename").read`
+Avoid: `File.read("filename")`
+Avoid: `IO.read("filename")`
+Use: `file("filename")`
+
+Similar to the command interactions these files will only be read localy
+with Ruby's internal calls. If you run this test against a remote target it won't
+read the file from the remote endpoint, but from the local OS instead.
+Use the `file` resource to read files on the target system.
+
+In general, try to avoid Ruby's IO calls from within InSpec controls and
+use InSpec resources instead.
+
+
+### 7. Avoid Ruby gem dependencies in controls
+
+In addition to avoiding system-level gems and modules you should also limit
+the use of external dependencies to resource packs or plugins. Gems need to be
+resolved, installed, vendored, and protected from conflicts. We aim to avoid
+exposing this complexity to users of InSpec, to make it a great tool even if
+you are not a developer.
+
+Developers may still use external gem dependencies but should vendor it
+with their plugins or resource packs.
+
+
+### 8. Avoid debugging calls (in production)
+
+Reason: One of the best way to develop and explore tests is the interactive debugging shell `pry` (see the section on "Interactive Debugging with Pry" at the end of this page). However, after you finish your profile make sure you have no interactive statements included anymore. Sometimes interactive calls are hidden behind conditionals (`if` statements) that are harder to reach. These calls can easily cause trouble when an automated profiles runs into an interactive `pry` call that stops the execution and waits for user input.
+
+Avoid: `binding.pry` in production profiles
+Use: Use debugging calls during development only
```

Comment:
Should we give an `Inspec::Log` example?

Reply:
Good point!